plugins/database/mongodb/connection_producer.go

```diff
@@ -106,120 +135,49 @@ func (c *mongoDBConnectionProducer) Init(ctx context.Context, conf map[string]in
 
 // Connection creates or returns an existing a database connection. If the session fails
 // on a ping check, the session will be closed and then re-created.
-func (c *mongoDBConnectionProducer) Connection(_ context.Context) (interface{}, error) {
+func (c *mongoDBConnectionProducer) Connection(ctx context.Context) (interface{}, error) {
 	if !c.Initialized {
 		return nil, connutil.ErrNotInitialized
 	}
 
-	if c.session != nil {
-		if err := c.session.Ping(); err == nil {
-			return c.session, nil
+	if c.client != nil {
+		if err := c.client.Ping(ctx, readpref.Primary()); err == nil {
+			return c.client, nil
 		}
-		c.session.Close()
+		// Ignore error on purpose since we want to re-create a session
+		_ = c.client.Disconnect(ctx)
```

Comment:
Should an error here be logged, or is it expected?

Reply:
It is expected since we cannot log here. We could only return an error which we obviously don't want.